### ISSUE TITLE
Android download maps crashes/stalls

### ISSUE THREAD

**Issue by zmichels** [StraboSpot/strabo-mobile#164]
When downloading map tiles for offline use on Android tablet:
Everything works fine for some zooms...
But stalls and needs to restart the app when downloading inner zooms from a region (totaling ~ 1400 tiles). I do not have this trouble with iOS devices.

**Comment by zmichels** [comment]
Workaround seems to be to open task manager on Android tablet when StraboMobile stalls on tile download. Then clicking on app in Task Manager returns to the app map screen.

**Comment by ginxwar** [comment]
I can test this out.  Maybe localforage has an update that we could look at again.  It would be great if we could use indexedDB as default and fallback to WebSQL if indexedDB isn't present

**Comment by ginxwar** [comment]
I made a change to the way offline tiles are stored.  Previously, both Android and iOS were forced using WebSQL as their storage unit.  WebSQL is kinda deprecated and isn't likely going to get standards support and IndexedDB is really the way to go.

I'm now defaulting the localforage library to choose the best available driver for that platform.  In iOS, that's WebSQL, because iOS doesn't support IndexedDB.  On Android, the best is IndexedDB.

I'm hoping IndexedDB on Android yield better performance. (It should).  I was able to save more than 2000 tiles on my Android device and it seemed to work ok.

I'll let @jessica-azgs review this and let her decide before she pulls it in the next build.

**Comment by jessica-azgs** [comment]
@ginxwar I did pull this into the build and it seems to be working for zooms 15 or greater when downloading inner zooms (which is what we have the user restricted to anyway). It's working for me as well if I start from zoom 14 and download inner zooms (about 1400 tiles) however the progress bar doesn't seem to show the actual progress. The process doesn't actually finish until a bit after the progress bar finishes so it may seem like it stalled to a user. Maybe there is a way to make the progress bar more accurate? If possible we might be able to allow the user to download starting at zoom 14, or lower even.

Also, I noticed that if you push the back button after some tiles have been downloaded, but before the process has actually finished, then those tiles that were downloaded show up on the Offline Maps page in the total number of tiles. However there is no list entry for the (incomplete) map they were supposed to go to so there no way to delete just those tiles.
